ios/RNSnowplowTracker.m

```diff
@@ -14,175 +14,216 @@ - (dispatch_queue_t)methodQueue
-          [self.self.tracker.subject setIpAddress:options[@"ipAddress"]];
-      }
-      if (options[@"useragent"] != nil) {
-          [self.self.tracker.subject setUseragent:options[@"useragent"]];
-      }
-      if (options[@"networkUserId"] != nil) {
-          [self.self.tracker.subject setNetworkUserId:options[@"networkUserId"]];
-      }
-      if (options[@"domainUserId"] != nil) {
-          [self.self.tracker.subject setDomainUserId:options[@"domainUserId"]];
-      }
+RCT_EXPORT_METHOD(setSubjectData :(NSDictionary *)options
+                                :rejecter:(RCTPromiseRejectBlock)reject) {
+
+    // ios throws if it finds an NSNull here - unset the field if null passed
+    if (options[@"userId"] != nil) {
+        if (options[@"userId"] == (id)[NSNull null]) {
+            [self.tracker.subject setUserId:nil];
+        } else {
+            [self.tracker.subject setUserId:options[@"userId"]];
+        }
+    }
+
+    if (options[@"timezone"] != nil) {
+        if (options[@"timezone"] == (id)[NSNull null]) {
+            [self.tracker.subject setTimezone:nil];
+        } else {
+            [self.tracker.subject setTimezone:options[@"timezone"]];
+        }
+    }
+
+    if (options[@"language"] != nil) {
+        if (options[@"language"] == (id)[NSNull null]) {
+              [self.tracker.subject setLanguage:nil];
+        } else {
+              [self.tracker.subject setLanguage:options[@"language"]];
+        }
+    }
+
+    if (options[@"ipAddress"] != nil) {
+        if (options[@"ipAddress"] == (id)[NSNull null]) {
+            [self.tracker.subject setIpAddress:nil];
+        } else {
+            [self.tracker.subject setIpAddress:options[@"ipAddress"]];
+        }
+    }
+    if (options[@"useragent"] != nil) {
+        if (options[@"useragent"] == (id)[NSNull null]) {
+            [self.tracker.subject setUseragent:nil];
+        } else {
+            [self.tracker.subject setUseragent:options[@"useragent"]];
+        }
+    }
+
+    if (options[@"networkUserId"] != nil) {
+        if (options[@"networkUserId"] == (id)[NSNull null]) {
+            [self.tracker.subject setNetworkUserId:nil];
+        } else {
+            [self.tracker.subject setNetworkUserId:options[@"networkUserId"]];
+        }
+    }
+
+    if (options[@"domainUserId"] != nil) {
+        if (options[@"domainUserId"] == (id)[NSNull null]) {
+            [self.tracker.subject setDomainUserId:nil];
+        } else {
+            [self.tracker.subject setDomainUserId:options[@"domainUserId"]];
+        }
+    }
+
+    if (options[@"screenWidth"] != nil && options[@"screenHeight"] != nil) {
+        if (options[@"screenWidth"] == (id)[NSNull null] || options[@"screenHeight"] == (id)[NSNull null]) {
+            NSError * error = [NSError errorWithDomain:@"SnowplowTracker" code:100 userInfo:nil];
+            reject(@"ERROR", @"SnowplowTracker: initiasetSubjectDatalize() method -  screenWidth and screenHeight cannot be null", error);
```

Comment:
I'm in doubt. Does `reject` stop the execution of the method too?
If no, does it make sense to return after reject?

Reply:
reject returns a rejected promise, and yes stops execution.

Reply:
After some testing, I was actually wrong - on configuring it with multiple error-invoking values, it attempts to reject multiple times, and crashes. I'll change it to return on error.